spring-grpc-core/src/main/java/org/springframework/grpc/server/ServletGrpcServerFactory.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright 2024-2025 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.grpc.server;
+
+import io.grpc.Server;
+import io.grpc.ServerServiceDefinition;
+
+/**
+ * Marker interface for {@link GrpcServerFactory} that is to be handled by the servlet
```

Comment:
Not super excited about this but the alternative is to pass in null in this case but then the filter implementations must take this into account and check for null and then assume null means servlet.